src/test/java/edu/oakland/test/database/DatabasePersistentStorageTest.java

```diff
@@ -0,0 +1,403 @@
+  }
+
+
+  @Test
+  @DisplayName("current Satellite matches next satellite")
+  void currentSatMatchNextSat() {
+    locDataPoint.add(ldp);
+    trackData.add(trackDataPoint);
+
+    String[] sampleSatArray = {"Sat1", "Sat2", "Sat3", "Sat4"};
+    DatabasePersistentStorageImplementation dpsc = new DatabasePersistentStorageImplementation(0, 
+        locDataPoint, 
+        trackData, 
+        sampleSatArray
+    );
+
+    assertEquals("Sat3", dpsc.getNextSat("Sat2"));
+  }
+
+  @Test
+  @DisplayName("cannot store null location data point")
+  void cannotStoreNullLocationDataPoint() {
+    locDataPoint.add(ldp);
+    trackData.add(trackDataPoint);
+
+    DatabasePersistentStorageImplementation dpsc = new DatabasePersistentStorageImplementation(0, 
+        locDataPoint, 
+        trackData, 
+        randomArray
+    );
+
+    assertThrows(IllegalArgumentException.class, () -> {
+      dpsc.storeLocationDataPoint(null);
+    });
+
+  }
+
+  @Test
+  @DisplayName("cannot store null track data")
+  void cannotStoreNullTrackData() {
+    locDataPoint.add(ldp);
+    trackData.add(trackDataPoint);
+
+    DatabasePersistentStorageImplementation dpsc = new DatabasePersistentStorageImplementation(0, 
+        locDataPoint, 
+        trackData, 
+        randomArray
+    );
+
+    assertThrows(IllegalArgumentException.class, () -> {
+      dpsc.storeTrackData(null);
+    });
+  }
+
+  @Test
+  @DisplayName("location data point offset out of bounds returns null")
+  void locationDataPointOffsetOutOfBoundsReturnsNull() {
+    locDataPoint.add(ldp);
+    trackData.add(trackDataPoint);
+
+    DatabasePersistentStorageImplementation dpsc = new DatabasePersistentStorageImplementation(0, 
+        locDataPoint, 
+        trackData, 
+        randomArray
+    );
+    dpsc.storeLocationDataPoint(ldp);
+    assertEquals(null, dpsc.getLocationDataPoint(99999));
+  }
+
+  @Test
+  //@DisplayName("track data offset out of bounds returns null")
+  //void trackDataOffsetOutOfBoundsIsReturnsNull() {
+  @DisplayName("track data offset out of bounds returns empty track data")
+  void trackDataOffsetOutOfBoundsIsReturnsEmptyTd() {
+    // Made changes per r532465218
+    // Not sure if we needed to change the DisplayName and func name.
+
+    // Setup
+    locDataPoint.add(ldp);
+    trackData.add(trackDataPoint);
+    DatabasePersistentStorageImplementation dpsc = new DatabasePersistentStorageImplementation(0, 
+            locDataPoint, 
+            trackData, 
+            randomArray
+        );
+
+    // UPDATE: We return an empty trackData now.
+    //TrackData randomTd = new TrackData(new LocationDataPoint[0]);
+    dpsc.storeTrackData(trackDataPoint);
+    // test if empty track data compared to empty track data.
+    TrackData testTrackData = dpsc.getTrackData(99999);
+    assertNotNull(testTrackData);
+  }
+
+  @Test
+  @DisplayName("track data offset out of bounds is invalid")
+  void trackDataOffsetOutOfBoundsIsInvalid() {
```

Comment:
Sorry, I messed up which methods were which. This can test all 3 assert statements.
```suggestion
  @DisplayName("track data offset out of bounds returns empty track data")
  void trackDataOffsetOutOfBoundsReturnsEmptyTrackData() {
```